AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/ReconcileAndLocalSaveOperation.swift

```diff
@@ -133,16 +133,21 @@ class ReconcileAndLocalSaveOperation: AsynchronousOperation {
 
         do {
             try storageAdapter.transaction {
-                queryPendingMutations(withModels: remoteModels.map(\.model))
+                self.queryLocalMetadata(remoteModels)
                     .subscribe(on: workQueue)
-                    .flatMap { mutationEvents -> Future<([RemoteModel], [LocalMetadata]), DataStoreError> in
-                        let remoteModelsToApply = self.reconcile(remoteModels, pendingMutations: mutationEvents)
-                        return self.queryLocalMetadata(remoteModelsToApply)
+                    .map { (remoteModels, localMetadatas) in
+                        self.getDispositions(for: remoteModels, localMetadatas: localMetadatas)
                     }
-                    .flatMap { (remoteModelsToApply, localMetadatas) -> Future<Void, DataStoreError> in
-                        let dispositions = self.getDispositions(for: remoteModelsToApply,
-                                                                localMetadatas: localMetadatas)
-                        return self.applyRemoteModelsDispositions(dispositions)
+                    .flatMap { dispositions in
+                        self.queryPendingMutations(withModels: dispositions.map(\.remoteModel.model))
+                            .map { pendingMutations in (pendingMutations, dispositions) }
+                    }
+                    .map { (pendingMutations, dispositions) in
+                        self.separateDispositions(pendingMutations: pendingMutations, dispositions: dispositions)
+                    }
+                    .flatMap { (dispositions, dispositionOnlyApplyMetadata) in
+                        self.waitAllPublisherFinishes(publishers: dispositionOnlyApplyMetadata.map(self.saveMetadata(disposition:)))
+                            .flatMap { _ in self.applyRemoteModelsDispositions(dispositions) }
                     }
```

Comment:
can you add doc comment for the high level details of this `reconcile(remoteModels)` method? So we have a clear understanding of what is logic (and the proposed in this PR) and why this order of operations is correct